optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/ChangeMove.java

```diff
@@ -105,28 +103,22 @@ public String getSimpleMoveTypeDescription() {
     }
 
     @Override
-    public boolean equals(Object o) {
+    public boolean equals(final Object o) {
         if (this == o) {
             return true;
-        } else if (o instanceof ChangeMove) {
-            ChangeMove<?> other = (ChangeMove) o;
-            return new EqualsBuilder()
-                    .append(entity, other.entity)
```

Comment:
Yea, yea, I know...
This already uses equals() instead of same semantics on entities, but in practice Move equality is only used for Move Tabu Search and nobody uses that. So this hasn't posed anywhere the trouble that VariableListenerNotification would have if you make that equals() semantics.
That being said, move equality should probably use same semantics. Can you create a separate JIRA for that? It will need to be thoroughly tested.